Commandline support

d3ming opened the pull request:
@tmillner 

This will complete https://github.com/paperg/tagcompare/issues/8.  Planning to bump version to 0.2.0 after this gets merged as it should be ready for wider adoption (to ad serving FT)

Coveralls for the build (slight increase to code coverage to 92.53%): https://coveralls.io/builds/4691059

d3ming committed:
Add commandline args support to tagcompare

Support commandline args, see README or run with --help to see supported commands.
Curently it says...

```
$ tagcompare --help
usage: tagcompare [-h] [--capture-only] [-d DOMAIN] [--verbose] [--version]
                  [-c CAMPAIGNS [CAMPAIGNS ...] | -p PUBLISHERS
                  [PUBLISHERS ...]]

optional arguments:
  -h, --help            show this help message and exit
  --capture-only        Run capture only without compare
  -d DOMAIN, --domain DOMAIN
                        Domain, i.e. www.placelocal.com
  --verbose             Enable verbose logging for debugging
  --version             Show version
  -c CAMPAIGNS [CAMPAIGNS ...], --campaigns CAMPAIGNS [CAMPAIGNS ...]
                        Space separated list of campaign ids
  -p PUBLISHERS [PUBLISHERS ...], --publishers PUBLISHERS [PUBLISHERS ...]
                        Space separated list of publisher ids
```

d3ming commented:
This will complete https://github.com/paperg/tagcompare/issues/8

tmillner commented on tagcompare/main.py:
```diff
@@ -1,10 +1,108 @@
+def __update_settings_from_args(args):
+    # Update settings based on params
+    if args.campaigns or args.publishers:
+        print args
```
Is this a remnant? Not sure why we'd need this.

tmillner commented on tagcompare/main.py:
```diff
@@ -1,10 +1,108 @@
+
     jobname = capture.main()
-    compare.main(build=jobname)
+    if not args.capture_only:
```
Preferential but I'd simplify this to make the argument --run-compare that way it's easier to read

d3ming replied:
I wanted it so that compare gets run by default

tmillner commented on tagcompare/placelocal.py:
```diff
@@ -128,7 +130,13 @@ def _get_all_pids(pids):
+def get_cids_from_settings():
+    cids = settings.DEFAULT.campaigns
+    pids = settings.DEFAULT.publishers
+    return _get_cids(cids, pids)
```
This is a bit magical, I'd expect (without seeing comments clarifying) this to simply return a list of cids (ie return line 134, cid)

tmillner commented on tagcompare/test/assets/test_settings.json:
```diff
@@ -6,7 +6,7 @@
     }
   },
-  "campaigns": [ ],
+  "campaigns": [ 131313 ],
```
This looks like a QA campaign?

d3ming replied:
It's just a number I put in for the unit test, we don't actually try to do anything with this campaign (so it doesn't matter if it exists or not)

tmillner commented on tagcompare/test/assets/test_settings.json:
```diff
@@ -6,7 +6,7 @@
   },
-  "campaigns": [ ],
+  "campaigns": [ 131313 ],
   "publishers": [ 627 ],
```
Which IHG is this?

tmillner commented:
Will this solve world hunger?

tmillner commented on README.md:
```diff
@@ -44,8 +44,16 @@ Update the `settings.local.json` file with the desired input to the tool as desi
 ## Running tests
 `make test` or `tox` will run unit tests
 `make test-all` will run all the tests
+`./test <path_to_test>` will run a specific test
```
If only a file path is passed in this will run a "set" of tests rather than a specific test, ex
 tagcompare/test/test_settings.py  will run 
- test_domain(),
- test_campagigns(), 
- test_publishers() 
- ...

d3ming replied:
Yeah I meant it will run every test in that file, will update

bitglue commented on setup.py:
```diff
@@ -30,7 +30,7 @@ def run_tests(self):
 setup(
     name='tagcompare',
-    version='0.1.1',
+    version='0.2.0-alpha2',
```
You might want to take a look at g-box's setup.py for a way to avoid manually updating this version number all the time.

d3ming replied:
Oh nice, does git version use tags to determine version?  If so, that's great!

kian commented on README.md:
```diff
@@ -44,8 +44,16 @@ Update the `settings.local.json` file with the desired input to the tool as desi
+Commandline arguments can optionally be used to configure the tagcompare run.
+`tagcompare --help` will show the current list of supported options.
+
+Note:Options passed through the commandline will override the `settings.local.json`
```
I'd remove "Note:" here

d3ming replied:
:+1:

kian commented on README.md:
```diff
@@ -44,8 +44,16 @@ Update the `settings.local.json` file with the desired input to the tool as desi
-All outputs generated by the tool will be stored in `/tmp/tagcompare`
+### Commandline arguments:
+Commandline arguments can optionally be used to configure the tagcompare run.
+`tagcompare --help` will show the current list of supported options.
```
:+1:

kian commented on README.md:
```diff
@@ -44,8 +44,16 @@ Update the `settings.local.json` file with the desired input to the tool as desi
 ## Running tests
 `make test` or `tox` will run unit tests
 `make test-all` will run all the tests
+`./test <path_to_test>` wills tests from a specific test file
```
missing a word - "will [run] tests"

d3ming replied:
Good catch

kian commented on setup.py:
```diff
@@ -28,9 +29,20 @@ def run_tests(self):
 setup(
     name='tagcompare',
-    version='0.1.1',
+    version=git_version(),
```
neat!

kian commented on tagcompare/main.py:
```diff
@@ -1,10 +1,115 @@
+LOGGER = logger.Logger(name="main", writefile=True).get()
+
+
+def __query_yes_no(question, default="yes"):
```
I'd probably look for a module w/ this functionality, though if no high quality ones exist, this looks fine

d3ming replied:
I looked, and this is what the internet came up with.  I took this code from (also noted in the method documentation): http://code.activestate.com/recipes/577058/

d3ming commented:
Thanks all for reviewing.  I should have addressed all the comments, let me know if there's anything you guys feel should block merge - otherwise I will merge by EOD
